CenterForOpenScience/osf.io
# [ENG-1941] Make Akismet Spam autobans reversible

## Purpose

When we ban someone, we want that activity to be reliably reversible. 

## Changes

- move ban behavior from admin view to model
- add tests for model methods


## QA Notes

- no migration
- low risk
- no permission code touched
- additive change.
- QA can verify be banning and unbanning users and nodes via the admin app and admin admin app.
- should effect Impact or proformance

## Documentation

🐞  fix, no docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-1941
